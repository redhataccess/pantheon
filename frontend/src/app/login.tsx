import React, { Component } from 'react';
import { Button, Alert, AlertActionCloseButton, BackgroundImage, BackgroundImageSrc, TextInput } from '@patternfly/react-core';
import '@app/app.css';

export default class Login extends Component {
  public state = {
    username: '',
    password: '',
    currentLogin: 'anonymous',
    authMessage: ''
  };

  public render() {
    const { username, password } = this.state;
    return (
      <React.Fragment>
        <div className="app-container">
          <div>
            {this.failedAuthMessage()}
            <TextInput id="username" type="text" placeholder="Username" value={username} onChange={this.onUsernameChange} />
            <TextInput id="password" type="password" placeholder="Password" value={password} onChange={this.onPasswordChange} />
            <div>
              {this.checkAuth()}
              <Button onClick={this.login}>Log In</Button>
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }

  failedAuthMessage = () => {
    return this.state.authMessage.length > 0 && <div className="notification-container">
      <Alert variant="danger"
          title={this.state.authMessage}
        action={<AlertActionCloseButton onClose={() => { this.setState({ failedAuth: false })}} />} />
    </div>
  }

  onUsernameChange = username => {
    this.setState({ username });

  };
  onPasswordChange = password => {
    this.setState({ password });
  };

  login = (postBody) => {
    const formData = new FormData();
    formData.append("j_username", this.state.username)
    formData.append("j_password", this.state.password)

    fetch('/j_security_check', {
      method: 'post',
      body: formData
    }).then(response => {
      if (response.status == 200) {
        console.log(" Works " + response.status)
        window.location.href = "/pantheon"
      } else if (response.status == 403) {
        this.setState({ authMessage: "Login failed, please try again." })
      } else {
        this.setState({ authMessage: "Unknown failure - HTTP " + response.status + ": " + response.statusText})
      }
    });
  }

  checkAuth = () => {
    console.log('Check auth: ' + this.state.currentLogin)
    if (this.state.currentLogin == 'anonymous') {
      fetch("/system/sling/info.sessionInfo.json")
        .then(response => response.json())
        .then(responseJSON => {
          if (responseJSON["userID"] != 'anonymous') {
            this.setState({ currentLogin: responseJSON["userID"] })
          }
        })
    }
  }
}
