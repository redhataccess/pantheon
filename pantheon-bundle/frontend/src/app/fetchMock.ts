if (!self.fetch) {
    self.fetch = (input: RequestInfo, init?: RequestInit | undefined) => {
        const p = new Promise<Response>((resolve, reject) => {
            // just don't do anything. this exists so that our tests don't try and actually resolve fetches,
            // (which is exactly what isomorphic-fetch tried to do while we were importing it...)
            // and calling nothing here allows us to skip all of our fetch.then() and fetch.catch() code.
        })
        return p
    }
}