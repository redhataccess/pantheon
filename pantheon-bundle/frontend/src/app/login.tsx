import React, { Component } from 'react';
import { Bullseye, Button, Alert, AlertActionCloseButton, FormGroup, TextInput } from '@patternfly/react-core';
import '@app/app.css';

class Login extends Component<any, any, any> {
  constructor(props) {
    super(props)
    this.state = {
      authMessage: '',
      password: '',
      username: ''
    };
  }

  public render() {
    return (
      <React.Fragment>
        <Bullseye>
          <div className="app-container">
            <div>
              {this.failedAuthMessage()}
              <FormGroup
                label="Username:"
                fieldId="username"
              >
                <TextInput id="username" type="text" placeholder="Username" value={this.state.username} onChange={this.onUsernameChange} onKeyPress={this.onLoginKeyPress} />
              </FormGroup>
              <br />
              <FormGroup
                label="Password:"
                fieldId="password"
              >
                <TextInput id="password" type="password" placeholder="Password" value={this.state.password} onChange={this.onPasswordChange} onKeyPress={this.onLoginKeyPress} />
              </FormGroup>
              <br />
              <div>
                <Button onClick={this.login}>Log In</Button>
              </div>
            </div>
          </div>
        </Bullseye>
      </React.Fragment>
    );
  }

  private onLoginKeyPress = (event) => {
    if (event.key === 'Enter') {
      this.login()
    }
  }

  private failedAuthMessage = () => {
    return this.state.authMessage.length > 0 && <div className="notification-container">
      <Alert variant="danger"
        title={this.state.authMessage}
        action={<AlertActionCloseButton onClose={this.resetAuthMessage} />} />
    </div>
  }

  private resetAuthMessage = () => {
    this.setState({ authMessage: '' })
  }

  private onUsernameChange = username => {
    this.setState({ username });

  };
  private onPasswordChange = password => {
    this.setState({ password });
  };

  private login = () => {
    const formData = new FormData();
    formData.append("j_username", this.state.username)
    formData.append("j_password", this.state.password)

    fetch('/j_security_check', {
      body: formData,
      method: 'post',
    }).then(response => {
      if (response.status === 200) {
        console.log(" Works " + response.status)
        window.location.href = "/pantheon"
      } else if (response.status === 403) {
        this.setState({ authMessage: "Login failed, please try again." })
      } else {
        this.setState({ authMessage: "Unknown failure - HTTP " + response.status + ": " + response.statusText })
      }
    });
  }
}

export { Login }