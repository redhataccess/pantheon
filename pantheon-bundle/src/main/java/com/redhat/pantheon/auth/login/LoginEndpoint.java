//package com.redhat.pantheon.auth.login;
//
//import org.apache.sling.api.SlingHttpServletRequest;
//import org.apache.sling.api.SlingHttpServletResponse;
//import org.apache.sling.api.servlets.SlingAllMethodsServlet;
//import org.osgi.service.component.annotations.Component;
//
//import javax.servlet.Servlet;
//import java.io.IOException;
//import java.net.URI;
//import java.net.URISyntaxException;
//import java.text.ParseException;
//import java.util.logging.Logger;
//
//@Component(
//        service = Servlet.class,
//        property = {
//                "sling.servlet.methods={GET, POST}",
//                "sling.servlet.paths=/sso/login"
//        },
//        immediate = true)
//public class LoginEndpoint extends SlingAllMethodsServlet {
//    private final Logger log = Logger.getLogger(LoginEndpoint.class.getName());
//
//    @Override
//    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
//
//            try {
//
//                URI uri = createAuthenticationRequest();
//
//                if (uri != null) {
//                    response.setStatus(302);
//                    response.setHeader("Location",uri.toString());
//
//                } else {
//                    log.info("[" + LoginEndpoint.class.getSimpleName() + "] Error occurred while creating the Authentication request.");
//                }
//
//            } catch (URISyntaxException e) {
//                log.info("Error occurred while creating the Authentication request.");
//            } catch (ParseException e) {
//                log.info("Error occurred while creating the Authentication request.");
//            }
//
//    }
//
//    private URI createAuthenticationRequest() throws URISyntaxException, ParseException {
//
//        return null;
//    }
//}
