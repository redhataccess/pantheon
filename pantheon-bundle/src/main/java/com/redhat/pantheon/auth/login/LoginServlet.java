package com.redhat.pantheon.auth.login;

import com.redhat.pantheon.auth.keycloak.KeycloakAuthenticationHandler;
import org.apache.sling.api.SlingHttpServletRequest;
import org.apache.sling.api.SlingHttpServletResponse;
import org.apache.sling.api.servlets.SlingAllMethodsServlet;
import org.osgi.service.component.annotations.Component;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.servlet.Servlet;
import java.io.IOException;
import java.net.URI;
import java.net.URISyntaxException;

import static com.redhat.pantheon.servlet.ServletUtils.writeAsJson;

/**
 * A login servlet that handles Keycloak or basic auth request
 *
 * @author Lisa Davidson
 */
@Component(
        service = Servlet.class,
        property = {
                "sling.servlet.methods={GET, POST}",
                "sling.servlet.paths=/auth/login"
                },
        immediate = true)
public class LoginServlet extends SlingAllMethodsServlet {
    private static final Logger log = LoggerFactory.getLogger(LoginServlet.class.getName());

    @Override
    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
        String uri = "";
        if (System.getenv("AUTH_SERVER_URL") != null) {
            try {
                uri = System.getenv("SSO_LOGIN_URL") != null ? System.getenv("SSO_LOGIN_URL") : "";

                if (uri.length() > 0) {
                    response.setStatus(302);
                    response.setHeader("Location", uri);
                } else {
                    log.error("Error occurred while creating the Authentication request.");
                }

            } catch (Exception e) {
                log.error("Error occurred while creating the Authentication request.");
            }
        } else {
            log.info("SSO is not enabled.");
            uri = "/pantheon/#/login";
            response.setStatus(302);
            response.setHeader("Location", uri);
        }
    }
}
