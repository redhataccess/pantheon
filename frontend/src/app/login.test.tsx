import React from 'react'
import { Login } from '@app/login'
import '@app/fetchMock'

import { mount, shallow } from 'enzyme'
import { Bullseye, TextInput, FormGroup, Button } from '@patternfly/react-core'

describe('Login tests', () => {
  test('should render Login component', () => {
    const view = shallow(<Login />)
    expect(view).toMatchSnapshot()
  })

  it('should render a form group', () => {
    const wrapper = mount(<Login />)
    const formGroup = wrapper.find(FormGroup)
    expect(formGroup.exists()).toBe(true)
  })

  it('should render a text input', () => {
    const wrapper = mount(<Login />)
    const textInput = wrapper.find(TextInput)
    expect(textInput.exists()).toBe(true)
  })

  it('should render a Button', () => {
    const wrapper = mount(<Login />)
    const button = wrapper.find(Button)
    expect(button.exists()).toBe(true)
  })

  it('should render a Bullseye layout', () => {
    const wrapper = mount(<Login />)
    const bullseyeLayout = wrapper.find(Bullseye)
    expect(bullseyeLayout.exists()).toBe(true)
  })

})
