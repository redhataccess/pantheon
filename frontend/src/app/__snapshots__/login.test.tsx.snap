// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests should render Login component 1`] = `
<Fragment>
  <Bullseye>
    <div
      className="app-container"
    >
      <div>
        <FormGroup
          fieldId="username"
          label="Username:"
        >
          <TextInput
            aria-label={null}
            className=""
            id="username"
            isDisabled={false}
            isReadOnly={false}
            isRequired={false}
            isValid={true}
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
        </FormGroup>
        <br />
        <FormGroup
          fieldId="password"
          label="Password:"
        >
          <TextInput
            aria-label={null}
            className=""
            id="password"
            isDisabled={false}
            isReadOnly={false}
            isRequired={false}
            isValid={true}
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </FormGroup>
        <br />
        <div>
          <Component
            onClick={[Function]}
          >
            Log In
          </Component>
        </div>
      </div>
    </div>
  </Bullseye>
</Fragment>
`;
